### DISCUSSION TITLE
How to induce latency in development environment?

### DISCUSSION THREAD

**Discussion by Quitalizner** [geckosio/geckos.io#241] [Help / Questions]
Colyseus provides a way of faking latency. We can set the latency to around 200ms which seems realistic. 
This could help us to implement stuff to provide a smooth gameplay when the player encounters latency.

Does geckos.io/server have any such methods?

**Comment by yandeu** [comment]
I guess you are looking for https://github.com/geckosio/snapshot-interpolation#about.

It can be used with or without geckos.

**Reply by Quitalizner** [reply]
No, I'm not taking about snapshot-interpolation.

But this here->
https://github.com/colyseus/colyseus/blob/master/packages/core/src/Server.ts#L205

Could be helpful to simulate latency in development environment

**Comment by arthuro555** [comment]
For windows, you could try this tool: https://github.com/jagt/clumsy

**Comment by yandeu** [comment]
Maybe warp the send and receive functions in `setTimeout()`?

**Reply by Quitalizner** [reply]
Hmm, yeah. That might work
